### PR TITLE
ddl: clean up cloud storage files after global sort

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #46258

### What is changed and how it works?

As the title said.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/backfilling_dispatcher.go [hunk 1]
```diff
@@ -489,6 +493,25 @@ func getSummaryFromLastStep(
 	return minKey, maxKey, totalKVSize, allDataFiles, allStatFiles, nil
 }
 
+func cleanupCloudStorageFiles(ctx context.Context, gTaskMeta *BackfillGlobalMeta) {
+	backend, err := storage.ParseBackend(gTaskMeta.CloudStorageURI, nil)
+	if err != nil {
+		logutil.Logger(ctx).Warn("cannot cleanup cloud storage files", zap.Error(err))
```
Comment (Benjamin2037) [inline ddl/backfilling_dispatcher.go:499]: After return error，will we retry later？

Comment (tangenta) [reply ddl/backfilling_dispatcher.go:499]: No.

Comment (D3Hunter) [inline ddl/backfilling_dispatcher.go:499]:
```suggestion
		logutil.Logger(ctx).Warn("failed to parse cloud storage url", zap.Error(err))
```

File: ddl/backfilling_dispatcher.go [hunk 2]
```diff
@@ -489,6 +493,25 @@ func getSummaryFromLastStep(
...
+		return
+	}
+	extStore, err := storage.New(ctx, backend, &storage.ExternalStorageOptions{})
+	if err != nil {
+		logutil.Logger(ctx).Warn("cannot cleanup cloud storage files", zap.Error(err))
```
Comment (D3Hunter) [inline ddl/backfilling_dispatcher.go:504]:
```suggestion
		logutil.Logger(ctx).Warn("failed to create cloud storage", zap.Error(err))
```

File: ddl/backfilling_dispatcher.go [hunk 3]
```diff
@@ -489,6 +493,25 @@ func getSummaryFromLastStep(
...
+		return
+	}
+	prefix := strconv.Itoa(int(gTaskMeta.Job.ID))
+	err = external.CleanUpFiles(ctx, extStore, prefix, cleanExternalStoreFilesConcurrency)
```
Comment (D3Hunter) [inline ddl/backfilling_dispatcher.go:508]: maybe add a integration test for it, see https://github.com/D3Hunter/tidb/blob/2c722e80c21971cd354ce9a774ea017755df29b1/tests/realtikvtest/importintotest4/global_sort_test.go#L59

File: ddl/backfilling_dispatcher.go [hunk 4]
```diff
@@ -489,6 +493,25 @@ func getSummaryFromLastStep(
 	return minKey, maxKey, totalKVSize, allDataFiles, allStatFiles, nil
 }
 
+func cleanupCloudStorageFiles(ctx context.Context, gTaskMeta *BackfillGlobalMeta) {
+	backend, err := storage.ParseBackend(gTaskMeta.CloudStorageURI, nil)
+	if err != nil {
+		logutil.Logger(ctx).Warn("failed to parse cloud storage url", zap.Error(err))
```
Comment (Benjamin2037) [inline ddl/backfilling_dispatcher.go:499]:
```suggestion
		logutil.Logger(ctx).Warn("failed to parse cloud storage uri", zap.Error(err))
```

File: ddl/backfilling_dispatcher.go [hunk 5]
```diff
@@ -433,7 +433,11 @@ func getRangeSplitter(
 	if err != nil {
 		return nil, err
 	}
-	extStore, err := storage.New(ctx, backend, &storage.ExternalStorageOptions{})
+	opt := &storage.ExternalStorageOptions{}
+	if intest.InTest {
+		opt.NoCredentials = true
+	}
+	extStore, err := storage.New(ctx, backend, opt)
```
Comment (D3Hunter) [inline ddl/backfilling_dispatcher.go:440]: seems we can put store init part into a small method

**Review by D3Hunter — APPROVED** [review]

File: ddl/backfilling_dispatcher.go [hunk 6]
```diff
@@ -493,6 +497,25 @@ func getSummaryFromLastStep(
 	return minKey, maxKey, totalKVSize, allDataFiles, allStatFiles, nil
 }
 
+func cleanupCloudStorageFiles(ctx context.Context, gTaskMeta *BackfillGlobalMeta) {
+	backend, err := storage.ParseBackend(gTaskMeta.CloudStorageURI, nil)
+	if err != nil {
+		logutil.Logger(ctx).Warn("failed to parse cloud storage uri", zap.Error(err))
```
Comment (lance6716) [inline ddl/backfilling_dispatcher.go:503]: is jobID already included in logger in ctx? If not I think we better log more details so manually deletion will be easier

Comment (tangenta) [reply ddl/backfilling_dispatcher.go:503]: No, but I think we can attach the jobID in next PR?

**Review by lance6716 — APPROVED** [review]

**Pull request merged**